### DISCUSSION TITLE
How to use current file name in QA

### DISCUSSION THREAD

**Discussion by brimwats** [chhoumann/quickadd#559] [Q&A]
I'd like to use the current file name, usually the name of a project, in the creation of a new note. 

For example I'm on a project called Thesis, and I want to make a note with the file name format of `2023-08-22--Thesis--Meeting` with the Meeting Template in the same folder as the current folder.

I've tried this a few ways and it doesn't work, but I think I'm misunderstanding. I've tried to create a template with file name format as `{{DATE}}--{{FIELD}} Meeting Minutes` in the hope that it would suggest fields like the documentation says. I've also tried `{{DATE}}--{{FIELD:alias}} Meeting Minutes`, I've also tried {{DATE}} {{NAME} (but def am using those wrong). 

I also tried making a macro that would copy the file name, run the meeting template and then editor command paste into the file name format of `{{DATE}}--{{VALUE}} `but no luck there as well. Any suggestions?

**Comment by jimbo108108** [comment]
I've got the same question. I'd like to make a macro that prompts for and creates the folder name and then creates the file with the same name in the folder.

**Comment by chhoumann** [comment]
Very late answer - sorry about that.

The token you were reaching for exists now: `{{FILENAMECURRENT}}` (shipped in QuickAdd 2.6.0) resolves to the active file's name. Your Thesis example becomes a Template choice with this File Name Format:

```
{{DATE}}--{{FILENAMECURRENT}}--Meeting
```

Run it while the Thesis note is open and you get `2023-08-22--Thesis--Meeting`. For the location, set the Template choice's folder mode to **Same folder as current file** and the note lands next to the active one.

@jimbo108108: folder paths accept format syntax too, and missing folders are created automatically. Use the same variable in the folder path and the File Name Format, and one prompt fills both:

- Folder: `Projects/{{VALUE:project}}`
- File Name Format: `{{VALUE:project}}`

Docs: https://quickadd.obsidian.guide/docs/FormatSyntax#filenamecurrent

Closing as resolved.

**Reply by brimwats** [reply]
thank-you!!
